ios/MaasCore/Sources/MaasComponents/Components/ButtonVariants.swift

```diff
@@ -0,0 +1,208 @@
+struct ColorfulButton: View {
+    
+    let colors: ButtonColors
+    let input: Button.InputType
+    
+    @Environment(\.isEnabled) var isEnabled
+    
+    var body: some View {
+        Button(
+            input.text,
+            icon: input.icon,
+            foregroundColor: colors.foreground.stateful(isActive: input.isLoading, isEnabled: isEnabled),
+            backgroundColor: colors.background.stateful(isActive: input.isLoading, isEnabled: isEnabled),
```

Comment:
So `Button.InputType` colors are completely ignored 🤔 I think they should take precedence and we only fall back to these colors as defaults.

Reply:
Keep in mind that if the SDK user specifies colors in `InputType`, they did so explicitly at the call-site, so it would be unexpected for them to not be respected.

Reply:
I left no colors in `init` for ButtonVariants

Reply:
Ah, OK, I see, so we offer no way to customize colors for `ButtonVariants`. Not obvious to me that we won't want to support custom colors for `PrimaryButton`, but seems sensible for now 👍